README.md

```diff
@@ -66,15 +66,82 @@ Also, value-objects are responsible for a little more than just plain data holdi
+  * This one was very interesting. First it showed me that the numeric operations must be taken out of the conversion 
+  service, so they will be unit tested easily regardless of the currencies.
+  * Second - I probably would have gone with the Money API which was added in Javax a while ago to not care about the 
+  money conversion code. I am not completely sure that this solution is 100% correct. It looks like it as far as I can 
+  see and from the tests done. The obvious thing which made the rounding errors appear in the first place was that the 
+  given amount did not have a scale set. But my solution will work perfect only for the money which have the subunit with 
+  2 decimals (cent / eurocent / ban / etc). I think that for the other types of currencies the scale must match the number
+  of subdivisions, case in which the rounding errors might return.
+  * For the purpose of this process I would say is a good enough solution, I will not change now the Money with the javax Money but in a real project I think that would be the best option 
+* 
+* `Value-object serialization`
+* Investigate whether it is possible to implement the value-object serialization, to avoid `value` nested field in JSON. 
+See [#10](https://github.com/cleankod/currency-rate-converter/pull/10) as a starting point. Or maybe there is a better solution to the problem at hand?
+  * Commit 2
+  * Yes, it is possible but I prefer a different approach in which the domain object are not tainted by serialization code
+  * That is why I fixed it with a simple serializer for each type with a value and it works like a charm (and the only change needed is in the app initializer)
+  * Did not like the changes needed to make the tests pass, it was a little tedious, however it works even though might be a cleaner way
+* 
 * Move parameter-specific logic outside the controller.
+  * DONE
+* 
 * Better error handling, especially of potential errors from NBP API.
+* 
 * Caching the NBP API results.
+  * Cached using the simplest local caching and explained how it can be properly done
 * Circuit-breaker for the NBP API client.
+  * Not implemented but commented out the code where it would be implemented with resilience4j
+  * 
 * Better logging with traceability.
+  * I added some logs mostly to test the caching. However, to achieve traceability we would need to be able to correlate the logs,
+  * associate all the logs belonging to a request. And a way to achieve this would be with a correlation id - unique per request.
+  * It would work very well with the account id or the account number if it is not a sensitive data and is safe to print it in the logs.
+  * The chosen unique identifier can pe pushed in the NDC context and the logger can be configured to include that parameter in the log pattern.
```

Comment:
How to achieve distributed tracing? Please desribe how you would approach such a thing